From 31f8eacb6269238e32a8ad4ddf83a64b7d77bd17 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Tue, 20 Dec 2022 11:40:58 +0200
Subject: [PATCH] [SW-115675] gaudi2: bis: fix efuse bit for bis

This patch corrects the efuse bit being used to
indicate BIS compliant. It is bit 25 instead of
bit 16 of efuse word 58. This patch uses already defined
macro EFUSE_ENABLE_BIS_COMPLIANT from emb-specs
header gaudi2_security.h

Change-Id: I8d99faafd4403fa12b5b05617f53f7d0a94594e9
---

diff --git a/arch/gaudi2/common/common.h b/arch/gaudi2/common/common.h
index d43d992..732d8a0 100644
--- a/arch/gaudi2/common/common.h
+++ b/arch/gaudi2/common/common.h
@@ -62,8 +62,6 @@
 #define PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_MODE_G3_CONF	0x5555
 #define PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_SEL_G4_CONF	0xFF
 
-#define EFUSE_SYSCFG_BIS_COMPLIANT_MODE		BIT(16)
-
 /*
  * TODO: These are temporary macros
  * and to be removed after GAUDI2 PO
diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index b08e2ee..99f38be 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -1511,5 +1511,5 @@
 	if (ret)
 		return true;
 
-	return !!(config_fuse & EFUSE_SYSCFG_BIS_COMPLIANT_MODE);
+	return !!(config_fuse & EFUSE_ENABLE_BIS_COMPLIANT);
 }
